stl/inc/chrono

```diff
@@ -2309,15 +2339,110 @@ namespace chrono {
     }
 #endif // __cpp_lib_concepts
 
+    // [time.zone.link]
+
+    class time_zone_link {
+    public:
+        time_zone_link(std::string_view _Name, std::string_view _Target) : _Name(_Name), _Target(_Target) {}
+
+        // FIXME: see comment in time_zone...
+        // time_zone_link(time_zone_link&&) = default;
+        // time_zone_link& operator=(time_zone_link&&) = default;
+
+        string_view name() const noexcept {
+            return _Name;
+        }
+
+        string_view target() const noexcept {
+            return _Target;
+        }
+
+    private:
+        std::string _Name;
+        std::string _Target;
+    };
+
+    _NODISCARD inline bool operator==(const time_zone_link& _Left, const time_zone_link& _Right) noexcept {
+        return _Left.name() == _Right.name();
+    }
+
+#ifdef __cpp_lib_concepts
+    _NODISCARD inline strong_ordering operator<=>(const time_zone_link& _Left, const time_zone_link& _Right) noexcept {
+        return _Left.name() <=> _Right.name();
+    }
+#endif // __cpp_lib_concepts
+
     // [time.zone.db]
 
-    // TRANSITION: work in progress
-    // STRUCT tzdb
+    _NODISCARD inline string _Xtzdb_generate_current_zone() {
+        unique_ptr<__std_tzdb_current_zone_info, decltype(&__std_tzdb_delete_current_zone)> _Info{
+            __std_tzdb_get_current_zone(), &__std_tzdb_delete_current_zone};
+        if (_Info == nullptr) {
+            _Xbad_alloc();
+        } else if (_Info->_Err == __std_tzdb_error::_Win_error) {
+            _XGetLastError();
+        } else if (_Info->_Err == __std_tzdb_error::_Icu_error) {
+            _Xruntime_error("FIXME: what generic message should I use here?");
+        }
+
+        return {_Info->_Tz_name};
+    }
+
+    template <class _Ty>
+    _NODISCARD const _Ty* _Locate_zone_impl(const std::vector<_Ty, _Crt_allocator<_Ty>>& _Vec, string_view _Name) {
+        const auto _Result = find_if(_Vec.begin(), _Vec.end(), [&](auto& _Tz) { return _Tz.name() == _Name; });
+        return _Result != _Vec.end() ? &(*_Result) : nullptr;
+    }
+
     struct tzdb {
+        vector<time_zone, _Crt_allocator<time_zone>> time_zones;
+        vector<time_zone_link, _Crt_allocator<time_zone_link>> links;
         vector<leap_second, _Crt_allocator<leap_second>> leap_seconds;

```

Comment:
Oh, that's surprising - the Standard depicts `std::vector` with `std::allocator` for elements, so `_Crt_allocator` is observable. Hmm, I wonder if we're forced to choose between technical non-conformance, or allocations showing up as "leaks".

Reply:
I suspect we should follow the Standard, and if someone asks about “leaks”, we’ll show them the mem-checkpoint API. (We can continue to use `_Crt_allocator` for purely internal allocations.)

Reply:
I will leave the `_Crt_allocator` on `leap_seconds` because this is why allocator stuff was moved into `xtzdb.h`. Should we file an issue or project card to revert this?